plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/PlainTextKafkaAdminFactory.java

```diff
@@ -14,45 +14,43 @@
 
 package io.trino.plugin.kafka;
 
-import io.trino.spi.HostAddress;
+import io.trino.plugin.kafka.security.ForKafkaPlainText;
 import io.trino.spi.connector.ConnectorSession;
 import org.apache.kafka.common.security.auth.SecurityProtocol;
 
 import javax.inject.Inject;
 
 import java.util.Properties;
-import java.util.Set;
 
+import static com.google.common.base.Preconditions.checkState;
 import static java.util.Objects.requireNonNull;
-import static java.util.stream.Collectors.joining;
 import static org.apache.kafka.clients.CommonClientConfigs.SECURITY_PROTOCOL_CONFIG;
-import static org.apache.kafka.clients.consumer.ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG;
 
 public class PlainTextKafkaAdminFactory
         implements KafkaAdminFactory
 {
-    private final Set<HostAddress> nodes;
+    private final KafkaAdminFactory delegate;
     private final SecurityProtocol securityProtocol;
 
     @Inject
     public PlainTextKafkaAdminFactory(
+            @ForKafkaPlainText KafkaAdminFactory delegate,
             KafkaConfig kafkaConfig,
             KafkaSecurityConfig securityConfig)
     {
+        this.delegate = requireNonNull(delegate, "delegate is null");
         requireNonNull(kafkaConfig, "kafkaConfig is null");
         requireNonNull(securityConfig, "securityConfig is null");
+        checkState(securityConfig.getSecurityProtocol().isPresent(), "Security protocol is missing");
 
-        nodes = kafkaConfig.getNodes();
-        securityProtocol = securityConfig.getSecurityProtocol();
+        securityProtocol = securityConfig.getSecurityProtocol().get();
```

Comment:
Instead of `Optional#get` we can use ` Optional#orElseThrow`

Reply:
And at other places too.

Reply:
Since I removed the `PlaintextKafka...Factory.java` as per https://github.com/trinodb/trino/pull/8743#discussion_r815782018, this is not required anymore.